internal/controller/services/auth/auth_controller_actions.go

```diff
@@ -143,3 +147,51 @@ func managePermissions(ctx context.Context, rr *odhtypes.ReconciliationRequest)
 
 	return nil
 }
+
+func createUserGroup(ctx context.Context, rr *odhtypes.ReconciliationRequest, userGroupName string) error {
+	namespace, err := actions.ApplicationNamespace(ctx, rr)
+	if err != nil {
+		logf.Log.Error(err, "error getting application namespace")
+		return err
+	}
+	userGroup := &userv1.Group{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: userGroupName,
+			// Otherwise it errors with  "error": "an empty namespace may not be set during creation"
+			Namespace: namespace,
+		},
+		// Otherwise is errors with "error": "Group.user.openshift.io \"odh-admins\" is invalid: users: Invalid value: \"null\": users in body must be of type array: \"null\""}
+		Users: []string{},
+	}
+
+	err = rr.Client.Create(ctx, userGroup)
+	if err != nil && !k8serr.IsAlreadyExists(err) {
+		return err
+	}
+
+	return nil
+}
+
+func createDefaultGroup(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	ok, err := isDefaultAuthMethod(ctx, rr)
+	if err != nil {
+		return err
+	}
+	if !ok {
+		logf.Log.Info("default auth method is not enabled")
+		return nil
+	}
+
+	release := cluster.GetRelease()
+	if release.Name == cluster.ManagedRhoai {
```

Comment:
should this check on SelfManaged?

Reply:
Good catch, thank you! Yes, I'll add the missing logic to include SelfManaged.